### PR TITLE
Mods to MYNN sfc and PBL for fractional/coupled

### PR DESCRIPTION
**Pull request by benwgreen**
MYNN surface layer needs very slight changes if it will ever be used to compute surface heat/moisture fluxes (currently done by sfc_nst, sfc_ocean, sfc_sice, sfc_drv_ruc) in the fractional framework.
MYNN boundary layer needs an updated way to compute fluxes for the ocean in the case of fractional mask (this change follows the procedure done in GFS_PBL_generic.F90).
Testing finds these changes make no difference in the case of non-fractional mask and atmosphere-only, so regression test GSD_v0 should pass.
@joeolson42 is in favor of these changes.
Refer to Issue 39 (https://github.com/NOAA-GSD/ccpp-physics/issues/39)

Suggested reviewers: @joeolson42 and @shansun6

### CODE REVIEW

**Review by DomHeinzeller — COMMENTED** [review]
Thanks! Looks good to me. I'll be testing this after #38 and #36 are merged.

**Comment by DomHeinzeller** [comment]
This PR has been pulled into https://github.com/NOAA-GSD/ccpp-physics/pull/41 and will be merged as part of it.

File: physics/module_MYNNPBL_wrapper.F90 [hunk 1]
```diff
@@ -541,6 +543,45 @@ SUBROUTINE mynnedmf_wrapper_run(        &
 !        wspd(i)=wind(i)
       enddo
 
+      ! BWG: Coupling insertion
+      if (cplflx) then
+        do i=1,im
+          if (oceanfrac(i) > zero) then ! Ocean only, NO LAKES
+            if (fice(i) > one - epsln) then ! no open water, use results from CICE
```
Comment (ShanSunNOAA) [inline physics/module_MYNNPBL_wrapper.F90:550]: How about changing if (fice(i) > one -epsln)  to if (.not. wet(i)), so it won't depend on epsln?

Comment (climbfuji) [reply physics/module_MYNNPBL_wrapper.F90:550]: @benwgreen please have a look. Thanks.

**Review by ShanSunNOAA — COMMENTED** [review]
Could you modify similar line in GFS_PBL_generic.F90
      if (fice(i) > one - epsln) then ! no open water, use results from CICE
epsln is not used and can be removed. Thanks!

**Comment by climbfuji** [comment]
> Could you modify similar line in GFS_PBL_generic.F90
> if (fice(i) > one - epsln) then ! no open water, use results from CICE
> epsln is not used and can be removed. Thanks!

This will impact the s2s runs when we merge this to the authoritative repositories. Is this what we want?

**Comment by ShanSunNOAA** [comment]
Yes, practically it is the same, but using if (.not. wet(i)) would be more
consistent with the rest of the routine. Thanks,  -Shan

**Comment by DomHeinzeller** [comment]
> > Could you modify similar line in GFS_PBL_generic.F90
> > if (fice(i) > one - epsln) then ! no open water, use results from CICE
> > epsln is not used and can be removed. Thanks!
> 
> This will impact the s2s runs when we merge this to the authoritative repositories. Is this what we want?

Shan, I can't find this line in GFS_PBL_generic.F90. That file in gsd/develop has:
```
!  --- ...  coupling insertion

      if (cplflx) then
        do i=1,im
          if (oceanfrac(i) > zero) then ! Ocean only, NO LAKES
            if ( .not. wet(i)) then ! no open water
              if (flag_cice(i)) then !use results from CICE
                dusfci_cpl(i) = dusfc_cice(i)
                dvsfci_cpl(i) = dvsfc_cice(i)
                dtsfci_cpl(i) = dtsfc_cice(i)
                dqsfci_cpl(i) = dqsfc_cice(i)
              else !use PBL fluxes when CICE fluxes is unavailable
                dusfci_cpl(i) = dusfc1(i)
                dvsfci_cpl(i) = dvsfc1(i)
                dtsfci_cpl(i) = dtsfc1(i)
                dqsfci_cpl(i) = dqsfc1(i)
              end if
            elseif (icy(i) .or. dry(i)) then ! use stress_ocean from sfc_diff for opw component at mixed point
              tem1 = max(q1(i), 1.e-8)
              rho = prsl(i,1) / (rd*t1(i)*(one+fvirt*tem1))
              if (wind(i) > zero) then
                tem = - rho * stress_wat(i) / wind(i)
                dusfci_cpl(i) = tem * ugrs1(i)   ! U-momentum flux
                dvsfci_cpl(i) = tem * vgrs1(i)   ! V-momentum flux
              else
                dusfci_cpl(i) = zero
                dvsfci_cpl(i) = zero
              endif
              dtsfci_cpl(i) = cp   * rho * hflx_wat(i) ! sensible heat flux over open ocean
              dqsfci_cpl(i) = hvap * rho * evap_wat(i) ! latent heat flux over open ocean
            else                                       ! use results from PBL scheme for 100% open ocean
              dusfci_cpl(i) = dusfc1(i)
              dvsfci_cpl(i) = dvsfc1(i)
              dtsfci_cpl(i) = dtsfc1(i)*hffac(i)
              dqsfci_cpl(i) = dqsfc1(i)*hefac(i)
            endif
!
            dusfc_cpl (i) = dusfc_cpl(i) + dusfci_cpl(i) * dtf
            dvsfc_cpl (i) = dvsfc_cpl(i) + dvsfci_cpl(i) * dtf
            dtsfc_cpl (i) = dtsfc_cpl(i) + dtsfci_cpl(i) * dtf
            dqsfc_cpl (i) = dqsfc_cpl(i) + dqsfci_cpl(i) * dtf
!
          else
            dusfc_cpl(i) = huge
            dvsfc_cpl(i) = huge
            dtsfc_cpl(i) = huge
            dqsfc_cpl(i) = huge
!!
          endif ! Ocean only, NO LAKES
        enddo
      endif
```
I will make the corresponding change in module_MYNNPBL_wrapper.F90 and push it to the PR.

**Comment by DomHeinzeller** [comment]
> > > Could you modify similar line in GFS_PBL_generic.F90
> > > if (fice(i) > one - epsln) then ! no open water, use results from CICE
> > > epsln is not used and can be removed. Thanks!
> > 
> > 
> > This will impact the s2s runs when we merge this to the authoritative repositories. Is this what we want?
> 
> Shan, I can't find this line in GFS_PBL_generic.F90. That file in gsd/develop has:
> 
> ```
> !  --- ...  coupling insertion
> 
>       if (cplflx) then
>         do i=1,im
>           if (oceanfrac(i) > zero) then ! Ocean only, NO LAKES
>             if ( .not. wet(i)) then ! no open water
>               if (flag_cice(i)) then !use results from CICE
>                 dusfci_cpl(i) = dusfc_cice(i)
>                 dvsfci_cpl(i) = dvsfc_cice(i)
>                 dtsfci_cpl(i) = dtsfc_cice(i)
>                 dqsfci_cpl(i) = dqsfc_cice(i)
>               else !use PBL fluxes when CICE fluxes is unavailable
>                 dusfci_cpl(i) = dusfc1(i)
>                 dvsfci_cpl(i) = dvsfc1(i)
>                 dtsfci_cpl(i) = dtsfc1(i)
>                 dqsfci_cpl(i) = dqsfc1(i)
>               end if
>             elseif (icy(i) .or. dry(i)) then ! use stress_ocean from sfc_diff for opw component at mixed point
>               tem1 = max(q1(i), 1.e-8)
>               rho = prsl(i,1) / (rd*t1(i)*(one+fvirt*tem1))
>               if (wind(i) > zero) then
>                 tem = - rho * stress_wat(i) / wind(i)
>                 dusfci_cpl(i) = tem * ugrs1(i)   ! U-momentum flux
>                 dvsfci_cpl(i) = tem * vgrs1(i)   ! V-momentum flux
>               else
>                 dusfci_cpl(i) = zero
>                 dvsfci_cpl(i) = zero
>               endif
>               dtsfci_cpl(i) = cp   * rho * hflx_wat(i) ! sensible heat flux over open ocean
>               dqsfci_cpl(i) = hvap * rho * evap_wat(i) ! latent heat flux over open ocean
>             else                                       ! use results from PBL scheme for 100% open ocean
>               dusfci_cpl(i) = dusfc1(i)
>               dvsfci_cpl(i) = dvsfc1(i)
>               dtsfci_cpl(i) = dtsfc1(i)*hffac(i)
>               dqsfci_cpl(i) = dqsfc1(i)*hefac(i)
>             endif
> !
>             dusfc_cpl (i) = dusfc_cpl(i) + dusfci_cpl(i) * dtf
>             dvsfc_cpl (i) = dvsfc_cpl(i) + dvsfci_cpl(i) * dtf
>             dtsfc_cpl (i) = dtsfc_cpl(i) + dtsfci_cpl(i) * dtf
>             dqsfc_cpl (i) = dqsfc_cpl(i) + dqsfci_cpl(i) * dtf
> !
>           else
>             dusfc_cpl(i) = huge
>             dvsfc_cpl(i) = huge
>             dtsfc_cpl(i) = huge
>             dqsfc_cpl(i) = huge
> !!
>           endif ! Ocean only, NO LAKES
>         enddo
>       endif
> ```
> 
> I will make the corresponding change in module_MYNNPBL_wrapper.F90 and push it to the PR.

@shansun6 please check this commit I just pushed: https://github.com/NOAA-GSD/ccpp-physics/pull/41/commits/728c076a64f921670e8be18d1b0c0d79de6e3254

File: physics/module_MYNNPBL_wrapper.F90 (hunk 1 above)
Comment (benwgreen) [reply physics/module_MYNNPBL_wrapper.F90:550]: Sorry for late response. I am fine with Shan's suggested change.

**Pull request merged**